Microsoft.Toolkit.Uwp.UI.Controls.Input/RichSuggestBox/SuggestionChosenEventArgs.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.Toolkit.Deferred;
+using Windows.UI.Text;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Provides data for the <see cref="RichSuggestBox.SuggestionChosen"/> event.
+    /// </summary>
+    public class SuggestionChosenEventArgs : DeferredEventArgs
```

Comment:
Why would this one need to be deferred? In the TokenizingTextBox our add event is deferred as we're asking the dev to create an item from text (which could involve a look-up), but the RichSuggestBox only allows picking from the existing list that was already fetched from the other deferred requested event? (Would we want a way if the user types some arbitrary text to be able to pass that back in an event to make a 'general' token? Would that be this same event or a different one?)

Reply:
@michael-hawker Even though we have the requested item already, this item only makes sense to the dev. RichSuggestBox merely takes whatever object that is selected and stores it within a `RichSuggestToken` instance. It doesn't read the selected object at all. Therefore when the `SuggestionChosen` event is triggered, the dev also needs to provide a text representation of the token to be displayed in the document, by setting the `Text` property in the event args. So what user typed "@Al" gets realized to "@Alan" for example. As well as setting the desired token styling by changing the `Format` property.

In short, on a token selected, the control expects some inputs from the dev regarding how that specific token should look like. And because these decisions may be async, I set the event to be deferrable. I do verify the query again after waiting for the event to make sure that the query matches what is currently in the document. If they don't match, the token is discarded. https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/db0599a33a92c70b0db86ced50d1877e908f3d06/Microsoft.Toolkit.Uwp.UI.Controls.Input/RichSuggestBox/RichSuggestBox.cs#L577-L584

Now one alternative way to avoid this is to ask dev to provide `Text` and `Format` on `SuggestionsRequested` event. However, I think this approach is not really dev friendly as it forces the suggestion collection to have a fixed type. Making the event not deferred would clean up the code quite a bit and I am all in for it. Please let me know if you have any ideas on how to refactor out the deferred event.